src/WebJobs.Script.WebHost/Management/InstanceManager.cs

```diff
@@ -130,9 +146,19 @@ public async Task<string> ValidateContext(HostAssignmentContext assignmentContex
             HttpResponseMessage response = null;
             try
             {
-                var zipUrl = assignmentContext.ZipUrl;
+                _logger.LogInformation($"Will be using {assignmentContext.ZipUrl.EnvVar} app setting as zip url");
+
+                var zipUrl = assignmentContext.ZipUrl.Url;
                 if (!string.IsNullOrEmpty(zipUrl))
                 {
+                    // ScmRunFromPackage can be empty since it is a placeholder for the actual build zip
+                    // so don't do a HEAD call against it. Just check if it is a valid blob url.
+                    if (assignmentContext.ZipUrl.IsScmRunFromPackage())
+                    {
+                        bool isValid = assignmentContext.ZipUrl.IsValidBlobReference();
```

Comment:
I think we need latency metrics for this network call also.

Reply:
Thinking about this more - i don't know if we even need validation if SCM_.. appsetting is being used.

Value cannot be set by users. 
If it is an invalid blob it will throw an exception in ApplyContext which looks like the right thing to me.
if it is a valid empty blob we will specialize with empty app which is what we want.
If it is a valid non-empty blob we will specialize with actual contents.

Reply:
That seems like the right approach. We can always modify it if we want to open this setting up to users for some reason.